### ISSUE TITLE
Add User Lookup Method

### ISSUE THREAD

**Issue by mr-devs** [truthy/osometweet#3]
Something that worked well for pulling user-account data that handles rate-limits as well as time-dependent errors

```python
def connect_to_endpoint(oauth, params):
    """Downloads data from Twitter based on the `oauth` object passed and the
    `params` created with the `create_params()` function. 

    If time-dependent errors (429, 500, 503)

    """
    
    switch = True
    
    while switch:
        response = oauth.get("https://api.twitter.com/2/users", params=params)
        
        # Get number of requests left with our tokens
        remaining_requests = int(response.headers["x-rate-limit-remaining"])
        
        # If that number is one, we get the reset-time and wait until then, plus 15 seconds.
        # This is caught below as well, however, we want to program defensively, if possible.
        if remaining_requests == 1:
            buffer_wait_time = 15 
            resume_time = dt.fromtimestamp( int(response.headers["x-rate-limit-reset"]) + buffer_wait_time )
            print(f"Waiting on Twitter.\n\tResume Time: {resume_time}")
            pause.until(resume_time)

        """To be safe, we check explicitly for these TIME DEPENDENT errors.
        That is, these errors can be solved simply by waiting a little while 
        and pinging Twitter again. So that's what we do."""
        if response.status_code != 200:

            # Too many requests error
            if response.status_code == 429:
                buffer_wait_time = 15 
                resume_time = dt.fromtimestamp( int(response.headers["x-rate-limit-reset"]) + buffer_wait_time )
                print(f"Waiting on Twitter.\n\tResume Time: {resume_time}")
                pause.until(resume_time)

            # Twitter internal server error
            elif response.status_code == 500:
                # Twitter needs a break, so we wait 30 seconds
                resume_time = dt.now().timestamp() + 30
                print(f"Waiting on Twitter.\n\tResume Time: {resume_time}")
                pause.unit(resume_time)

            # Twitter service unavailable error
            elif response.status_code == 503:
                # Twitter needs a break, so we wait 30 seconds
                resume_time = dt.now().timestamp() + 30
                print(f"Waiting on Twitter.\n\tResume Time: {resume_time}")
                pause.unit(resume_time)

            # If we get this far, we've done something wrong and should exit
            raise Exception(
                "Request returned an error: {} {}".format(
                    response.status_code, response.text
                )
            )
        
        # If we get a 200 response, lets exit the function and return the response.json
        if response.ok:
            return response.json()
```

@ChrisTorresLugo this would require that I touch both #2 and #1. I don't need to touch #1 however this endpoint is much more effective with the user-based Oauth1 method.

**Comment by ChrisTorresLugo** [comment]
Ah, you already have some code for that. Go ahead. 

When I wrote #2, I was thinking of the 3-legged process.

**Comment by mr-devs** [comment]
This is addressed in pull request #4. Examples of using the method can be found in the [examples folder](https://github.com/truthy/osometweet/tree/master/examples).
